tchannel/serializer/thrift.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from thrift.protocol import TBinaryProtocol
+from thrift.transport import TTransport
+
+from tchannel.schemes import THRIFT
+
+from .. import io
+from .. import rw
+
+
+class ThriftSerializer(object):
+
+    _headers_rw = rw.headers(
+        rw.number(2),
+        rw.len_prefixed_string(rw.number(2)),
+        rw.len_prefixed_string(rw.number(2)),
+    )
+
+    def __init__(self, deserialize_type):
+        self.name = THRIFT
+        self.deserialize_type = deserialize_type
+
+    def serialize_header(self, headers):
+        result = self._headers_rw.write(headers, io.BytesIO()).getvalue()
+        return result
+
+    def deserialize_header(self, headers):
+        headers = io.BytesIO(headers)
+        headers = self._headers_rw.read(headers)
+        result = dict(headers)
+
+        return result
+
+    def serialize_body(self, call_args):
+
+        # TODO - use fastbinary directly
```

Comment:
I asked for fastbinary to be used because this is how the generated code looks for `read/write`:

    def write(self, oprot):
      if oprot.__class__ == TBinaryProtocol.TBinaryProtocolAccelerated and self.thrift_spec is not None and fastbinary is not None:
        oprot.trans.write(fastbinary.encode_binary(self, (self.__class__, self.thrift_spec)))
        return
      [...]

    def read(self, iprot):
      if iprot.__class__ == TBinaryProtocol.TBinaryProtocolAccelerated and isinstance(iprot.trans, TTransport.CReadableTransport) and self.thrift_spec is not None and fastbinary is not None:
        fastbinary.decode_binary(self, iprot.trans, (self.__class__, self.thrift_spec))
        return
      [...]

Instantiating the buffer, the protocol, etc. is unnecessary work when we can just do that.